### PR TITLE
include: xen: add libxenvchan.h header

### PR DESCRIPTION
**Pull request by sa-kib**
Import from Xen project public headers w/o modification.

### CODE REVIEW

File: include/zephyr/xen/public/io/libxenvchan.h [hunk 1]
```diff
@@ -0,0 +1,101 @@
+/**
```
Comment (firscity) [inline include/zephyr/xen/public/io/libxenvchan.h:1]: Zephyr license style requires single line SPDX Identifier at the start of the file:
`/* SPDX-License-Identifier: MIT */`

To find an example, please refer to other public headers in mainline.

File: include/zephyr/xen/public/io/libxenvchan.h [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+ *  path.
+ *
+ *  The ring.h macros define an asymmetric interface to a shared data structure
+ *  that assumes all rings reside in a single contiguous memory space. This is
+ *  not suitable for vchan because the interface to the ring is symmetric except
+ *  for the setup. Unlike the producer-consumer rings defined in ring.h, the
+ *  size of the rings used in vchan are determined at execution time instead of
+ *  compile time, so the macros in ring.h cannot be used to access the rings.
+ */
+
```
Comment (firscity) [inline include/zephyr/xen/public/io/libxenvchan.h:49]: Please, add a header guards here (`#ifndef __XEN_PUBLIC_IO_LIBENVCHAN_H__ ...`)

**Review by firscity — CHANGES_REQUESTED** [review]
Also, please add more information to commit message - why we need it and exact version of Xen, where you took this header from (hope it is a stable version, because it may vary in the middle between stable version, I recommend to use 4.17.0)

File: include/zephyr/xen/public/io/libxenvchan.h (hunk 1 above)
Comment (sa-kib) [reply include/zephyr/xen/public/io/libxenvchan.h:2]: done

File: include/zephyr/xen/public/io/libxenvchan.h (hunk 2 above)
Comment (sa-kib) [reply include/zephyr/xen/public/io/libxenvchan.h:32]: done

**Comment by sa-kib** [comment]
> Also, please add more information to commit message - why we need it and exact version of Xen, where you took this header from (hope it is a stable version, because it may vary in the middle between stable version, I recommend to use 4.17.0)

commit msg updated

**Review by firscity — APPROVED** [review]
`Reviewed-by: Dmytro Firsov <dmytro_firsov@epam.com>`

**Pull request merged**